apps/prairielearn/assets/scripts/behaviors/bootstrap-compat.ts

```diff
@@ -0,0 +1,557 @@
+import { observe } from 'selector-observer';
```

Comment:
Note to self: `selector-observer` doesn't react to class names changing on existing elements. For instance, if we have an element that's present when the page is loaded with `mr-2`, and the class list is later overwritten, we have no way to react to that. This is visible on the course instance billing page: load it up, check/uncheck a checkbox, and observe that the margin next to the progress bar disappears:

<img width="285" alt="Screenshot 2024-08-07 at 16 48 41" src="https://github.com/user-attachments/assets/351e91f3-4210-4404-ab2d-9627e4b34083">

Reply:
We'll probably have to do something custom with `MutationObserver` and `selector-set`.